hedera-node/src/main/java/com/hedera/services/ledger/TransferLogic.java

```diff
@@ -74,6 +82,8 @@ public class TransferLogic {
     private final TransactionalLedger<
                     Pair<AccountID, TokenID>, TokenRelProperty, MerkleTokenRelStatus>
             tokenRelsLedger;
+    private final TransactionContext txnCtx;
```

Comment:
`AutoCreationLogic` already has these dependencies (`txnCtx` + `aliasManager`); and `AutoCreationLogic` already has impure state `pendingCreations` that is being `reset()` with each usage.

To avoid creating the `tokenAliasMap` for every `CryptoTransfer`,  I'd lean toward putting in `AutoCreationLogic` e.g.:
```
private boolean aliasTokenUsagesAreKnown;
private Map<ByteString, Set<EntityNum>> aliasTokenUsages = new HashMap<>();

...

public void reset() {
    pendingCreations.clear();
    aliasTokenUsages.clear();
    aliasTokenUsagesAreKnown = false;
}

...
```
and then only creating this map on-demand.

Reply:
Agreed !